packages/react-meteor-data/suspense/useTracker.ts

```diff
@@ -129,14 +138,22 @@ export function useTrackerSuspenseNoDeps<T = any>(key: string, reactiveFn: IReac
     // Let subsequent renders know we are mounted (render is committed).
     refs.isMounted = true
 
-    // stop the computation on unmount
-    return () => {
-      if (refs.computation) {
-        refs.computation.stop()
-        delete refs.computation
-      }
+    // In strict mode (development only), `useEffect` may run 1-2 times.
+    // To avoid this, check the `timeout` to ensure cleanup only occurs after unmount.
+    if (refs.cleanupTimoutId) {
+      clearTimeout(refs.cleanupTimoutId)
+      delete refs.cleanupTimoutId
```

Comment:
Using the `delete` operator in V8 (the engine behind Node.js and Chrome) is generally not good for performance, especially in long-running or high-throughput applications.

When you use `delete obj.key`, V8 must change the object’s internal structure . This invalidates V8’s optimizations and forces the engine to fall back to slower property storage mechanisms .

Set it to undefined instead

refs:
https://phillcode.hashnode.dev/javascript-delete
https://alexanderobregon.substack.com/p/what-happens-when-you-delete-properties


Run this benchmark

<img width="697" height="249" alt="Image" src="https://github.com/user-attachments/assets/26464d3e-2005-4f11-a31a-472f10f239a2" />